File: Swift/0003.swift
class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        let characters = Array(s)  // Convert the string to an array of characters
        var charSet = Set<Character>()  // This set will keep track of characters in the current window
        var left = 0  // Left pointer of the sliding window
        var maxLength = 0  // Maximum length of substring without repeating characters
        
        for right in 0..<characters.count {
            // Move the right pointer and include the new character in the window
            while charSet.contains(characters[right]) {
                // Remove characters from the left side of the window until the current character is unique
                charSet.remove(characters[left])
                left += 1
            }
            charSet.insert(characters[right])
            maxLength = max(maxLength, right - left + 1)  // Update the maximum length
        }
        
        return maxLength
    }
}

File: C++/0003.cpp
#include <unordered_map>
#include <string>
#include <algorithm>

class Solution {
public:
    int lengthOfLongestSubstring(std::string s) {
        std::unordered_map<char, int> charMap;
        int maxLength = 0;
        int left = 0;

        for (int right = 0; right < s.length(); ++right) {
            char currentChar = s[right];

            // If the character is found in the map and it's within the current window
            if (charMap.find(currentChar) != charMap.end() && charMap[currentChar] >= left) {
                // Move the left pointer to the right of the last occurrence of currentChar
                left = charMap[currentChar] + 1;
            }

            // Update the last seen index of the current character
            charMap[currentChar] = right;

            // Calculate the length of the current substring and update maxLength
            maxLength = std::max(maxLength, right - left + 1);
        }

        return maxLength;
    }
};
